fuel-relayer/README.md

```diff
@@ -34,96 +28,25 @@ There are few rules that `da_height` (da as data availability) need to follow an
-
-### Syncing flow
-
-Tricky thing that can happen not so often is in initial synchronization. When do we start log subscription and when do we stop importing log from block. Initial sync is done as:
-
-1. sync from HardCoddedContractCreatingBlock to (BestEthBlock-EthFinalitySlider)
-1. sync overlap from LastIncludedEthBlock to BestEthBlock and save them in dequeue.
-1. Start listening to eth events
-1. Check if our LastIncludedEthBlock is same as BestEthBlock.
-  If not the same, stop listening to events and do 2,3,4 steps again.
-1. Continue to active listen on eth events. and prune(commit to db) dequeue for older finalized events
-
-To have validity of ValidatorSet we need to have logs from ethereum to reconstruct the set. So It does not matter for us if we start syncing to fuel network if we are not connected to eth client. And in case if fuel-bft ask for validator set of not yet synced relayer, we are going to return error. We probably need to have notification when relayer gets fully synced.
-
-On active sync when initial sync is finished we are watching new eth events and new chained block:
-For log watch we are doing:
-
-1. Receive log event:
-    1. if it is removal, append them into pending_removal vec
-    1. if it is new event. Apply pending_removal_vec and add new event to dequeue.
-1. On new fuel block, check finality and commit changed to db.
-    1. just to be sure introduce check of events by calling eth block logs. This is a double check just to be sure that we match with that is inside contract.
-
-For passive sync we are using ethereum pubsub protocol to get `logs` event: <https://geth.ethereum.org/docs/rpc/pubsub> . Example of eth log from log subscription:
-
-* "address":0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,
-* "topics":[0x00,0x00,0x00],
-* "data":"Bytes",
-* "block_hash":Some(Hash)),
-* "block_number":Some(14069822),
-* "transaction_hash":Some(0xcf7baaa6e6cd2d863807a3b8d3168f8d2699ad043e3ac249e82887fda6c4ef55),
-* "transaction_index":Some(130),
-* "log_index":Some(172),
-* "transaction_log_index":"None",
-* "log_type":"None",
-* "removed":"Some(false)"
-
-## Block Commitments
-
-When new fuel block is made it needs to be committed to ethereum network.
-For every block that we don't have fuel commit we try to do our best guess and send block that contracts can include, gaps in height will be discarded by contract.
-
-There is timeframe of few eth blocks where current leader can broadcast block to ethereum network. In case that he do it in time he will receive additional reward. In case that leader does not commit fuel block next leader will have ability to do it for him as bundle block commit, in that case bonus reward will go to second leader that has spend additional gas to push this change to ethereum network. This is still WIP discussion.
-
-Github discussion on fuel sync with L1: <https://github.com/FuelLabs/fuel-specs/issues/285>
+## Contract Flake
```

Comment:
Is the Fuel Nix flake also being introduced in this PR? Would it be possible to break that out into a separate PR for cleaner review?

Reply:
I have moved the nix stuff too https://github.com/FuelLabs/fuel-core/pull/665